cmd/notary/keys.go

```diff
@@ -599,7 +575,11 @@ func (k *keyCommander) keyPassphraseChange(cmd *cobra.Command, args []string) er
 	// unlocking passphrase and reusing that.
 	passChangeRetriever := k.getRetriever()
 	keyStore, err := trustmanager.NewKeyFileStore(config.GetString("trust_dir"), passChangeRetriever)
-	err = keyStore.AddKey(filepath.Join(keyGUN, keyID), role, privKey)
+	keyInfo, err := cs.GetKeyInfo(keyID)
+	if err != nil {
+		return err
+	}
+	err = keyStore.AddKey(privKey, keyInfo)
```

Comment:
Good catch!  This is tricky, I've pushed a way of doing this but it isn't very clean because we have to make a new keystore with a new passphrase retriever so that we overwrite the previous key